Add support for the new categorize_and_describe endpoint

brettmortensen opened the pull request:
@film42 @newellista

brettmortensen committed:
Add support for the new categorize_and_describe endpoint

liveh2o commented on src/main/java/atrium/AtriumClient.java:
```diff
@@ -577,6 +577,28 @@ public Account readAccount(String userGUID, String accountGUID) {
+   * @param transactions Required. An array of RawTransaction objects.
+   * @return CategorizedTransaction[]. This returns an array of CategorizedTransaction objects.
+   */
+  public CategorizedTransaction[] categorizeAndDescribeTransactions(JsonArray transactions) {
```
What's the thinking behind returning a different object type here? Is there something that the `CategorizedTransaction` provides that the `Transaction` does not?

brettmortensen replied:
Thanks for pointing that out. I hadn't realized it was a subset. Removed `CategorizedTransaction` in favor of `Transaction`.

film42 commented on src/main/java/atrium/AtriumClient.java:
```diff
@@ -577,6 +577,28 @@ public Account readAccount(String userGUID, String accountGUID) {
+   * @param transactions Required. An array of RawTransaction objects.
+   * @return CategorizedTransaction[]. This returns an array of CategorizedTransaction objects.
+   */
+  public CategorizedTransaction[] categorizeAndDescribeTransactions(JsonArray transactions) {
```
Why a JsonArray and not something like `List<Transaction> transactions`?

brettmortensen replied:
Updated. This approach is the way to go. 👍 

I originally went with `JsonArray` as it was the convention we were using in a few other endpoints in this library. I'll be getting up another PR soon to update the other endpoints. (https://github.com/mxenabled/atrium-java/pull/5)

Commit:
Update categorize_and_describe endpoint

Remove CategorizedTransaction model, in favor of the Transaction model
as it was a subset. This will be used as the return type. Use List over
JsonArray for transactions parameter in categorize_and_describe endpoint
for simplicity.

film42 commented on src/main/java/atrium/AtriumClient.java:
```diff
@@ -577,6 +578,29 @@ public Account readAccount(String userGUID, String accountGUID) {
+   * @param transactions Required. An array of RawTransaction objects.
+   * @return Transaction[]. This returns an array of Transaction objects.
+   */
+  public Transaction[] categorizeAndDescribeTransactions(List<Transaction> transactions) {
```
Response should probably be a List type.

brettmortensen commented:
This PR is no longer necessary with https://github.com/mxenabled/atrium-java/pull/7
